packages/flutter/lib/src/widgets/decorated_sliver.dart

```diff
@@ -0,0 +1,89 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'image.dart';
+
+/// A sliver widget that paints a [Decoration] either before or after its child
+/// paints.
+///
+/// Unlike [DecoratedBox], this widget expects its child to be a sliver, and
+/// must be placed in a widget that expects a sliver.
+///
+/// If the child sliver has infinite [SliverGeometry.scrollExtent], then we only
+/// draw the decoration down to the bottom [SliverGeometry.cacheExtent], and
+/// it is necessary to ensure that the bottom border does not creep
+/// above the top of the bottom cache.
+///
+/// Commonly used with [BoxDecoration].
+///
+/// The [child] is not clipped. To clip a child to the shape of a particular
+/// [ShapeDecoration], consider using a [ClipPath] widget.
```

Comment:
Does this mean users cannot provide a ShapeDecoration to DecoratedSliver.decoration? Or should not?

Reply:
They could but it won't clip the contents of the child. I guess the developer should try to guarantee that the contents of the child fits within the decoration.